BridgeAppSDK/SBAScheduledActivityManager.swift

```diff
@@ -140,27 +141,36 @@ public class SBAScheduledActivityManager: NSObject, SBASharedInfoController, ORK
         
-        let expiredKey = "expiredOn"
+        // syoung 05/06/2016 As of Swift 2.2, there is no equivilent to NSStringFromSelector
+        let expiredKey = "expiresOn"
         let scheduledKey = "scheduledOn"
         let finishedKey = "finishedOn"
+        let optionalKey = "persistent"
+        
+        // For unscheduled we need to also filter out those tasks that were finished prior to today
+        let finishedToday = NSPredicate(day: NSDate(), dateKey: finishedKey)
+        let unfinished = NSPredicate(format: "%K == nil", finishedKey)
+        let optionalFilter = NSPredicate(format: "%K == 1", optionalKey)
+        let unfinishedOrFinishedToday = NSCompoundPredicate(orPredicateWithSubpredicates: [unfinished, finishedToday])
+        let expiredYesterday = NSPredicate(day: NSDate().dateByAddingTimeInterval(-24*60*60), dateKey: expiredKey)
+        let notExpired = NSCompoundPredicate(notPredicateWithSubpredicate: expiredYesterday)
+        let notOptional = NSCompoundPredicate(notPredicateWithSubpredicate: optionalFilter)
+        let calendar = NSCalendar.currentCalendar()
+        let tomorrow = calendar.startOfDayForDate(NSDate().dateByAddingTimeInterval(24*60*60))
+        let notFuture = NSPredicate(format: "%K == nil OR %K < %@", scheduledKey, scheduledKey, tomorrow)
         
         switch section {
             
-        case .Yesterday:
-            // only expired schedules for yesterday
-            return NSPredicate(day: NSDate().dateByAddingTimeInterval(-24*60*60), dateKey: expiredKey)
+        case .ExpiredYesterday:
+            // expired yesterday section only showns those expired tasks that are also unfinished
+            return NSCompoundPredicate(andPredicateWithSubpredicates: [unfinished, expiredYesterday])
             
         case .Today:
-            // scheduled for today only
-            return NSPredicate(day: NSDate(), dateKey: scheduledKey)
+            return NSCompoundPredicate(andPredicateWithSubpredicates: [notOptional, notFuture, unfinishedOrFinishedToday, notExpired])
             
-        case .Unscheduled:
-            // For unscheduled we need to also filter out those tasks that were finished prior to today
-            let finishedToday = NSPredicate(day: NSDate(), dateKey: finishedKey)
-            let unfinished = NSPredicate(format: "%K == nil", finishedKey)
-            let unscheduled = NSPredicate(format: "%K == nil", scheduledKey)
-            let unfinishedOrFinishedToday = NSCompoundPredicate(orPredicateWithSubpredicates: [unfinished, finishedToday])
-            return NSCompoundPredicate(andPredicateWithSubpredicates: [unscheduled, unfinishedOrFinishedToday])
+        case .KeepGoing:
+            // Keep going section includes optional tasks that are either unfinished or were finished today
+            return NSCompoundPredicate(andPredicateWithSubpredicates: [optionalFilter, notFuture,unfinishedOrFinishedToday, notExpired])
```

Comment:
Keep going section should only include unfinished optional tasks, I think; the server will reschedule them when done so they'll always be there, and if we also show the ones they finished today, they'll be duplicated.